libbpf-tools/ttysnoop.c

```diff
@@ -0,0 +1,221 @@
+/* SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause) */
+
+/*
+ * ttysnoop     Watch live output from a tty or pts device.
+ *
+ * Copyright (c) 2021 Hengqi Chen
+ *
+ * Based on ttysnoop(8) from BCC by Brendan Gregg
+ * 11-Sep-2021   Hengqi Chen   Created this.
+ */
+#include <argp.h>
+#include <errno.h>
+#include <signal.h>
+#include <string.h>
+#include <time.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include <bpf/libbpf.h>
+#include <bpf/bpf.h>
+#include "ttysnoop.h"
+#include "ttysnoop.skel.h"
+#include "trace_helpers.h"
+
+#define PERF_BUFFER_PAGES	64
+#define PERF_POLL_TIMEOUT_MS	100
+#define warn(...) fprintf(stderr, __VA_ARGS__)
+
+static volatile sig_atomic_t exiting = 0;
+
+static bool clear_screen = true;
+static __u64 pts = -1;
+static int read_bytes = 256;
+static int read_count = 16;
+
+const char *argp_program_version = "ttysnoop 0.1";
+const char *argp_program_bug_address =
+	"https://github.com/iovisor/bcc/tree/master/libbpf-tools";
+const char argp_program_doc[] =
+"Watch live output from a tty or pts device.\n"
+"\n"
+"USAGE: ttysnoop [-h] [-C] [-s SIZE] [-c COUNT] DEVICE\n"
+"\n"
+"EXAMPLES:\n"
+"    ttysnoop /dev/pts/2    # snoop output from /dev/pts/2\n"
+"    ttysnoop 2             # snoop output from /dev/pts/2 (shortcut)\n"
+"    ttysnoop /dev/console  # snoop output from the system console\n"
+"    ttysnoop /dev/tty0     # snoop output from /dev/tty0\n"
+"    ttysnoop 2 -s 1024     # snoop output from /dev/pts/2 with data size 1024\n"
+"    ttysnoop 2 -c 2        # snoop output from /dev/pts/2 with 2 checks for 256"
+" bytes of data in buffer (potentially retrieving 512 bytes)\n";
+
+static const struct argp_option opts[] = {
+	{ "noclear", 'C', NULL, 0, "Don't clear the screen" },
+	{ "datasize", 's', "SIZE", 0, "size of the transmitting buffer (default 256)" },
```

Comment:
this is called `'data-size`` in `datacount`'s docstr

Reply:
i see what you mean about the bcc version. I agree that dropping the option is the best approach here.